pkgs/stdenv/darwin/default.nix

```diff
@@ -59,11 +59,12 @@ in rec {
   stageFun = step: last: {shell             ? "${bootstrapTools}/bin/bash",
                           overrides         ? (self: super: {}),
                           extraPreHook      ? "",
+                          extraNativeBuildInputs,
                           extraBuildInputs,
```

Comment:
@dezgeg There's still a notion of needing dependencies at build time or run time, and eventually `buildInputs` should not appear on the PATH to make this clear.